MAUI/MauiAppWithBroker/MSALClient/PCAWrapper.cs

```diff
@@ -0,0 +1,105 @@
+        private const string ClientId = "858b4a09-dc31-45d3-83a7-2b5f024f99cd"; // TODO - Replace with your client Id. And also replace in the AndroidManifest.xml
+
+        // TenantID of the organization (ms sample testing)
+        private const string TenantId = "7f58f645-c190-4ce5-9de4-e2b7acd2a6ab"; // TODO - Replace with your TenantID. And also replace in the AndroidManifest.xml
+
+        /// <summary>
+        /// Scopes defining what app can access in the graph
+        /// </summary>
+        internal static string[] Scopes = { "User.Read" };
+
+        // private constructor for singleton
+        private PCAWrapper()
+        {
+            // Create PCA once. Make sure that all the config parameters below are passed
+            PCA = PublicClientApplicationBuilder
+                                        .Create(ClientId)
+                                        .WithTenantId(TenantId)
+                                        .WithLogging(LogHere)
+                                        .WithBroker()
+                                        .WithRedirectUri(PlatformConfig.Instance.RedirectUri)
+                                        .WithIosKeychainSecurityGroup("com.microsoft.adalcache")
+                                        .Build();
+        }
+
+        // This demos logging
+        private void LogHere(LogLevel level, string message, bool containsPii)
+        {
+            // You can do customized logging here. This is just for demo.
+            Debug.WriteLine(message);
+        }
+
+        /// <summary>
+        /// Acquire the token silently
+        /// </summary>
+        /// <param name="scopes">desired scopes</param>
+        /// <returns>Authentication result</returns>
+        internal async Task<AuthenticationResult> AcquireTokenSilentAsync(string[] scopes)
+        {
+            var accts = await PCA.GetAccountsAsync().ConfigureAwait(false);
+            var acct = accts.FirstOrDefault();
+
+            var silentParamBuilder = PCA.AcquireTokenSilent(scopes, acct);
+            var authResult = await silentParamBuilder
+                                        .ExecuteAsync().ConfigureAwait(false);
+            return authResult;
+
+        }
+
+        /// <summary>
+        /// Perform the interactive acquisition of the token for the given scope
+        /// </summary>
+        /// <param name="scopes">desired scopes</param>
+        /// <returns></returns>
+        internal async Task<AuthenticationResult> AcquireTokenInteractiveAsync(string[] scopes)
+        {
+            return await PCA.AcquireTokenInteractive(scopes)
+                                    .WithParentActivityOrWindow(PlatformConfig.Instance.ParentWindow)
+                                    .ExecuteAsync()
+                                    .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signout may not perform the complete signout as company portal may hold
+        /// the token.

```

Comment:
Can you clarify what this means? As in if I call RemoveAsync(acct), then call ATS, it might still return a token.

Reply:
This will remove the acct from MSAL.NET cache. But Android broker for company portal will still have it.